django/db/models/fields/generated.py

```diff
@@ -37,10 +37,12 @@ def __init__(self, *, expression, db_persist=None, output_field=None, **kwargs):
     def cached_col(self):
         from django.db.models.expressions import Col
 
-        return Col(self.model._meta.db_table, self.output_field)
+        return Col(self.model._meta.db_table, self, self.output_field)
```

Comment:
That was the crux of the issue behind [these errors](https://djangoci.com/job/pull-requests-focal/database=postgres,label=focal-pr,python=python3.11/21887/testReport/junit/model_fields.test_generatedfield/StoredGeneratedFieldTests/test_output_field_lookups/)